app/src/main/java/net/kwatts/powtools/adapters/RideListAdapter.java

```diff
@@ -104,4 +117,22 @@ void bind(RideRow rideRow) {
         }
     }
 
+    void loadTrack(long rideId, @NonNull ImageView imageView) {
+        App.dbExecute(database -> {
+            List<Pair<Double, Double>> track = new ArrayList<>();
+            List<Moment> moments = database.momentDao().getFromRide(rideId);
+            for (Moment moment : moments) {
+                // The standard ordering of geo coordinates (according to ISO 6709) is latitude then
+                // longitude, but we swap the ordering here as Android's canvas renders with it's
+                // axes labeled in x, y ordering (which is the ordering that TrackDrawable expects).
+                //
+                // https://en.wikipedia.org/wiki/ISO_6709
+                track.add(new Pair<>(moment.getGpsLongDouble(), moment.getGpsLatDouble()));
+            }
+
+            new Handler(Looper.getMainLooper()).post(() ->
+                    imageView.setImageDrawable(new TrackDrawable(track)));
```

Comment:
is it best practice to create a new Handler for every one of these or reuse one (or a pool)?

Reply:
Keeping a reference to a `Handler` as a field would be slightly more efficient. Laziness prevailed!